contract/src/test/kotlin/com/r3/corda/sdk/token/contracts/EvolvableTokenTests.kt

```diff
@@ -0,0 +1,719 @@
+            transaction {
+                output(TestEvolvableTokenContract.ID, TestEvolvableTokenTypeFactory.withOneMaintainerAndOneObserver())
+
+                // With create command signed by maintainer and observer
+                tweak {
+                    command(listOf(ALICE.publicKey, CHARLES.publicKey), Create())
+                    failsWith(expectedError)
+                }
+            }
+
+
+            // With 2 maintainers and 2 observers
+            transaction {
+                output(TestEvolvableTokenContract.ID, TestEvolvableTokenTypeFactory.withTwoMaintainersAndTwoObservers())
+
+                // With create command signed by both maintainers and first observer
+                tweak {
+                    command(listOf(ALICE.publicKey, BOB.publicKey, CHARLES.publicKey), Create())
+                    failsWith(expectedError)
+                }
+
+                // With create command signed by both maintainers and second observer
+                tweak {
+                    command(listOf(ALICE.publicKey, BOB.publicKey, DENISE.publicKey), Create())
+                    failsWith(expectedError)
+                }
+
+                // With create command signed by both maintainers and both observers
+                tweak {
+                    command(listOf(ALICE.publicKey, BOB.publicKey, CHARLES.publicKey, DENISE.publicKey), Create())
+                    failsWith(expectedError)
+                }
+            }
+        }
+
+        @Test
+        fun `may only have one command`() {
+            val expectedError = "Evolvable token transactions support exactly one command only."
+
+            // With 1 maintainer
+            transaction {
+                output(TestEvolvableTokenContract.ID, TestEvolvableTokenTypeFactory.withOneMaintainer())
+
+                // With 1 create command
+                tweak {
+                    command(ALICE.publicKey, Create())
+                    verifies()
+                }
+
+                // With 2 create commands from the same maintainer
+                // Note: Due to change in V4, the TransactionBuilder will coalesce commands by CommandData. This means
+                // that multiple commands of the same type are merged with the union of all signers.
+                // This will _not_ error.
+                tweak {
+                    command(ALICE.publicKey, Create())
+                    command(ALICE.publicKey, Create())
+                    verifies()
+                }
+
+                // With 1 create and 1 update command from the same maintainer
+                tweak {
+                    command(ALICE.publicKey, Create())
+                    command(ALICE.publicKey, Update())
+                    failsWith(expectedError)
+                }
+
+                // With 2 create commands from different parties
+                // Note: Due to change in V4, the TransactionBuilder will coalesce commands by CommandData. This means
```

Comment:
I think this is being reverted in C5

Reply:
Or even earlier in 4.1. I propose to add a guard around the test to change the expected response based on version. Will look into the right way to version sniff.

Reply:
Added TODO tags to track the behaviour in 4.1 or 5.0.